artifactory/services/go/publishzipandmod.go

```diff
@@ -74,16 +74,14 @@ func (pwa *publishZipAndModApi) PublishPackage(params GoParams, client *jfroghtt
 	return &utils.OperationSummary{TotalSucceeded: totalSucceed, TotalFailed: totalFailed, TransferDetailsReader: content.NewContentReader(tempFile, "files")}, nil
 }
 
-func uploadArchive(params GoParams, moduleId, ext, url string, filesDetails *[]clientutils.FileTransferDetails, pwa *publishZipAndModApi) (success, failed int, err error) {
+func uploadArchive(params GoParams, archivePath string, moduleId, ext, url string, filesDetails *[]clientutils.FileTransferDetails, pwa *publishZipAndModApi) (success, failed int, err error) {
```

Comment:
Since this function is not used for uploading archives only, but is is used for uploading the go.mod as well, I suggest renaming it to **uploadFile**.